### PR TITLE
Exclude unit tests when running `mypy` with `pre-commit`

### PR DESCRIPTION
**Pull request by FlorentClarret**
### What does this PR do?

Exclude unit tests when running `mypy` with `pre-commit`

### Motivation

We can't run linter on this folder yet because tghis is not a valid python package yet

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -28,6 +28,7 @@ repos:
         - types-requests==2.30.0
         - types-tabulate==0.9.0.20240106
         - types-toml==0.10.8.20240310
+      exclude: ^tasks/unit-tests/.*$
```
Comment (FlorentClarret) [inline .pre-commit-config.yaml:31]:
```suggestion
      exclude: ^tasks/unit_tests/.*$
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -28,6 +28,7 @@ repos:
         - types-requests==2.30.0
         - types-tabulate==0.9.0.20240106
         - types-toml==0.10.8.20240310
+      exclude: ^tasks/unit_tests/.*$
```
Comment (KevinFairise2) [inline .pre-commit-config.yaml:31]:
```suggestion
      exclude: ^tasks/unit-tests/.*$
```

**Review by KevinFairise2 — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 2 above)
Comment (FlorentClarret) [reply .pre-commit-config.yaml:31]: It got renamed last Friday to `unit_tests` 🙂

**Review by CelianR — APPROVED** [review]

**Pull request merged**